### ISSUE TITLE
Terraform plan plugin crash

### ISSUE THREAD

**Issue by jarriagad** [Telmate/terraform-provider-proxmox#512]
During the running of `terraform apply` the plugin stopped responding and got the listed at the bottom

I will continue troubleshooting this issue and will add any additional info as update to this issue

`Error: Plugin did not respond
│
│   with proxmox_vm_qemu.kube-cluster-1[0],
│   on main.tf line 17, in resource "proxmox_vm_qemu" "kube-cluster-1":
│   17: resource "proxmox_vm_qemu" "kube-cluster-1" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with proxmox_vm_qemu.salt-master[0],
│   on main.tf line 61, in resource "proxmox_vm_qemu" "salt-master":
│   61: resource "proxmox_vm_qemu" "salt-master" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v2.9.3 plugin:

panic: interface conversion: interface {} is nil, not []interface {}

goroutine 86 [running]:
github.com/Telmate/proxmox-api-go/proxmox.(*Client).GetVmInfo(0x1400040b770, 0x140005d3650, 0x1e, 0xd0, 0x140005b6900)
	github.com/Telmate/proxmox-api-go@v0.0.0-20211123192920-062fd1a6ab10/proxmox/client.go:159 +0x548
github.com/Telmate/terraform-provider-proxmox/proxmox._resourceVmQemuRead(0x14000180c00, 0x105432c00, 0x1400040b7c0, 0x0, 0x0)
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:1213 +0x264
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuRead(0x14000180c00, 0x105432c00, 0x1400040b7c0, 0x105578210, 0x0)
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:1190 +0x38
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x1400018a7e0, 0x105599158, 0x1400033c200, 0x14000180c00, 0x105432c00, 0x1400040b7c0, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:346 +0x170
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x1400018a7e0, 0x105599158, 0x1400033c200, 0x1400010a0d0, 0x105432c00, 0x1400040b7c0, 0x1400061cd40, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/resource.go:635 +0x160
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x1400012d278, 0x105599158, 0x1400033c200, 0x1400033c2c0, 0x1400033c200, 0x1052e3a4c, 0x10550fc00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.9.0/helper/schema/grpc_provider.go:576 +0x3ac
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x14000571440, 0x105599200, 0x1400033c200, 0x140004ca780, 0x14000571440, 0x1052a2200, 0x105515fa0)
	github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/tf5server/server.go:298 +0xe4
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler(0x10553de00, 0x14000571440, 0x105599200, 0x1400033e480, 0x140004ca720, 0x0, 0x105599200, 0x1400033e480, 0x140001d4000, 0x7f7)
	github.com/hashicorp/terraform-plugin-go@v0.4.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:344 +0x1c8
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002968c0, 0x1055a0eb8, 0x1400011ad80, 0x14000408360, 0x1400007e450, 0x1059cfd10, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.42.0/server.go:1282 +0x3e8
google.golang.org/grpc.(*Server).handleStream(0x140002968c0, 0x1055a0eb8, 0x1400011ad80, 0x14000408360, 0x0)
	google.golang.org/grpc@v1.42.0/server.go:1616 +0xa50
google.golang.org/grpc.(*Server).serveStreams.func1.2(0x140000a0190, 0x140002968c0, 0x1055a0eb8, 0x1400011ad80, 0x14000408360)
	google.golang.org/grpc@v1.42.0/server.go:921 +0x94
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.42.0/server.go:919 +0x1f8

Error: The terraform-provider-proxmox_v2.9.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.`

**Comment by jarriagad** [comment]
[apply.log](https://github.com/Telmate/terraform-provider-proxmox/files/8174245/apply.log)

Attaching failure log after re-running `terraform plan`

**Comment by jarriagad** [comment]
Upgraded provider to latest

Terraform v1.1.7
on darwin_arm64
+ provider registry.terraform.io/telmate/proxmox v2.9.6

I was able to get the job to run after rebooting my laptop and deleting the .terraform* objects in my main directory.

This issue seem to be random.

Let me know if i there is more information that I can provide

**Comment by mleone87** [comment]
post your .tf file please

**Comment by jarriagad** [comment]
main.tf:
```
terraform {
  required_providers {
    proxmox = {
      source = "telmate/proxmox"
      version = "2.9.6"
    }
  }
}

provider "proxmox" {
  #Prox API url
  pm_api_url = "https://server.domain.us:8006/api2/json"
  #tls insecure allow
  pm_tls_insecure = true
}

resource "proxmox_vm_qemu" "kube-cluster-1" {
  count       = 1
  name        = "kube-cluster-1"
  target_node = var.proxmox_host
  clone       = var.template_name
  full_clone  = true
  pool        = "devel"
  vmid        = 1101

  agent       = 0
  os_type     = "cloud-init"
  cores       = 4
  sockets     = 1
  cpu         = "host"
  memory      = 8000
  scsihw      = "virtio-scsi-pci"
  bootdisk    = "scsi0"

  disk {
    slot     = 0
    size     = "100G"
    type     = "scsi"
    storage  = "bulkData"
    iothread = 1
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.1.1.91/24,gw=10.1.1.1"
  nameserver = "10.1.1.31"

  sshkeys = <<EOF
  ${var.ssh_key}
  EOF
}
resource "proxmox_vm_qemu" "salt-master" {
  count       = 1
  name        = "salt-master01"
  target_node = var.proxmox_host
  clone       = var.template_name
  full_clone  = true
  pool        = "devel"
  vmid        = 1103

  agent       = 0
  os_type     = "cloud-init"
  cores       = 2
  sockets     = 1
  cpu         = "host"
  memory      = 4096
  scsihw      = "virtio-scsi-pci"
  bootdisk    = "scsi0"

  disk {
    slot     = 0
    size     = "35G"
    type     = "scsi"
    storage  = "bulkData"
    iothread = 1
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.1.1.93/24,gw=10.1.1.1"

  sshkeys = <<EOF
  ${var.ssh_key}
  EOF
}
```

vars.tf:
```variable "ssh_key" {
  default = "<rsa-key>"
}
variable "proxmox_host" {
    default = "server02"
}
variable "template_name" {
    default = "focal-minion-cloudinit-template"
}
```

kubernetes-full-cluster.tf:
```
resource "proxmox_vm_qemu" "kube-cluster-1-loadBalancers" {
  count       = 2
  name        = "kc1-lb${count.index + 1}"
  target_node = var.proxmox_host
  clone       = var.template_name
  full_clone  = true
  pool        = "devel"
  vmid        = "${count.index + 200}"

  agent       = 0
  os_type     = "cloud-init"
  cores       = 2
  sockets     = 1
  cpu         = "host"
  memory      = 2048
  scsihw      = "virtio-scsi-pci"
  bootdisk    = "scsi0"

  disk {
    slot     = 0
    size     = "20G"
    type     = "scsi"
    storage  = "bulkData"
    iothread = 1
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.1.1.4${count.index}/24,gw=10.1.1.1"

  sshkeys = <<EOF
  ${var.ssh_key}
  EOF
}



resource "proxmox_vm_qemu" "kube-cluster-1-controllers" {
  count       = 3
  name        = "kc1-controller-${count.index + 1}"
  target_node = var.proxmox_host
  clone       = var.template_name
  full_clone  = true
  pool        = "devel"
  vmid        = "${count.index + 202}"

  agent       = 0
  os_type     = "cloud-init"
  cores       = 2
  sockets     = 1
  cpu         = "host"
  memory      = 4096
  scsihw      = "virtio-scsi-pci"
  bootdisk    = "scsi0"

  disk {
    slot     = 0
    size     = "100G"
    type     = "scsi"
    storage  = "bulkData"
    iothread = 1
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.1.1.${count.index + 42}/24,gw=10.1.1.1"

  sshkeys = <<EOF
  ${var.ssh_key}
  EOF
}

resource "proxmox_vm_qemu" "kube-cluster-1-workers" {
  count       = 3
  name        = "kc1-worker-${count.index + 1}"
  target_node = var.proxmox_host
  clone       = var.template_name
  full_clone  = true
  pool        = "devel"
  vmid        = "${count.index + 205}"

  agent       = 0
  os_type     = "cloud-init"
  cores       = 2
  sockets     = 1
  cpu         = "host"
  memory      = 4096
  scsihw      = "virtio-scsi-pci"
  bootdisk    = "scsi0"

  disk {
    slot     = 0
    size     = "100G"
    type     = "scsi"
    storage  = "bulkData"
    iothread = 1
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.1.1.${count.index + 45}/24,gw=10.1.1.1"

  sshkeys = <<EOF
  ${var.ssh_key}
  EOF
}

resource "proxmox_vm_qemu" "kube-cluster-1-ingressController1" {
  count       = 1
  name        = "kc1-ingress-1"
  target_node = var.proxmox_host
  clone       = var.template_name
  full_clone  = true
  pool        = "devel"
  vmid        = 208

  agent       = 0
  os_type     = "cloud-init"
  cores       = 2
  sockets     = 1
  cpu         = "host"
  memory      = 2048
  scsihw      = "virtio-scsi-pci"
  bootdisk    = "scsi0"

  disk {
    slot     = 0
    size     = "20G"
    type     = "scsi"
    storage  = "bulkData"
    iothread = 1
  }

  network {
    model  = "virtio"
    bridge = "vmbr0"
  }

  lifecycle {
    ignore_changes = [
      network,
    ]
  }

  ipconfig0 = "ip=10.1.1.48/24,gw=10.1.1.1"

  sshkeys = <<EOF
  ${var.ssh_key}
  EOF
}
```

**Comment by mleone87** [comment]
where did you get this?
```
  lifecycle {
    ignore_changes = [
      network,
    ]
  }
```

other than that since you did not mention anything about credentials, pretty sure this is a permissions problem

**Comment by jarriagad** [comment]
I found that from some tutorial, I went ahead and removed it in case it was causing issues.

I think root cause is unrelated to Terraform and actually caused by an underlying hardrive issue on the hypervizor I am using.
The failing drives cause the mounted file system to go into read-only mode and don't allow Terraform to apply the changes needed. So I guess in a way, it was permissions :)

Thank you the help!
